### PR TITLE
Password checker including letters, numbers and special characters with 8 characters (Zeyu Chen)

### PR DESCRIPTION
**Pull request by Cikky0209**
Add a new feature to ensure  that all user's password contain letters, numbers and special characters to increase the password security

Also asking for a minimum length of 8 characters in signing up page.

### CODE REVIEW

File: core/settings.py [hunk 1]
```diff
@@ -186,9 +186,8 @@
 EMAIL_HOST = 'smtp.gmail.com'
 EMAIL_PORT = 587 # For TLS, 465 for SSL
 EMAIL_USE_TLS = True
-EMAIL_HOST_USER = 'deakinhardhatwebsite@gmail.com'
-EMAIL_HOST_PASSWORD = 'dtrv xywy rpqq tozw'
-
+EMAIL_HOST_USER = 'hardhatcompanywebsite@gmail.com'
+EMAIL_HOST_PASSWORD = 'nuje nbmo cfqe skjb'
```
Comment (s-bonnor) [inline core/settings.py:190]: App password for email account should not be stored in git

**Review by s-bonnor — CHANGES_REQUESTED** [review]
Password validation looks good
Security issue with storing the app password in git
also seems to be a few unrelated files included in this PR under pages/*

File: core/settings.py (hunk 1 above)
Comment (Cikky0209) [reply core/settings.py:190]: This is the code of the SMTP service, not the Password. The content displayed has no security implications. I also added additional content (including front-end and security aspects). Please check

**Comment by Cikky0209** [comment]
Resolve the conflict from secure_code file

**Review by s-bonnor — APPROVED** [review]
Looks good to me

File: core/settings.py (hunk 1 above)
Comment (s-bonnor) [reply core/settings.py:190]: I'm still sure that storing an app access token in the repository is a massive security issue but we can handle that at a later date

**Pull request merged**